### ISSUE TITLE
[Bug] Clicking twice on backdrop will corrupt the modal

### ISSUE THREAD

**Issue by kopax** [reactstrap/reactstrap#360]
### Steps to reproduce issue

__soft test__ (chrome)

- Go on http://reactstrap.github.io/components/modals/
- Open ReactTools tab in developer console.
- Find the modal in react dom
-  For closing, double click on the background.

__hard test__ (chrome)

- Go on http://reactstrap.github.io/components/modals/
- Open ReactTools tab in developer console.
- Find the modal in react dom
- Now repeat click on the `Open modal` Button like if you were in a Diablo III game

### Expected

multiple click on the backdrop should close the modal

### Result

It will create multiple instance of the modal.
Problematically, it means that if you try to reopen your modal in the `onExit` function, this will not work as the state get corrupted.

As a result it is not possible to force the modal to be open.

**Comment by TheSharpieOne** [comment]
This issue is similar to #251

**Comment by kopax** [comment]
Could it be possible to just use a `setState({ isInProgress })` and check weither the click should be ignored or not ?

**Comment by QuantumHive** [comment]
I'm using
 - `bootstrap 4.0.0-alpha.6`
 - `react 15.6.1`
 - `reactstrap 4.8.0`

And I'm having the same issue when doing a multiclick on backdrop getting the following error in the console when I try to reopen the modal:

```
invariant.js?c115:44 Uncaught Error: _registerComponent(...): Target container is not a DOM element.
    at invariant (invariant.js?c115:44)
    at Object._renderNewRootComponent (ReactMount.js?d3c5:310)
    at Object._renderSubtreeIntoContainer (ReactMount.js?d3c5:401)
    at Object.renderSubtreeIntoContainer [as unstable_renderSubtreeIntoContainer] (ReactMount.js?d3c5:342)
    at Modal.renderIntoSubtree (reactstrap.es.js?8ce4:2422)
    at Modal.hide (reactstrap.es.js?8ce4:2374)
    at Modal.togglePortal (reactstrap.es.js?8ce4:2351)
    at Modal.componentDidUpdate (reactstrap.es.js?8ce4:2286)
    at measureLifeCyclePerf (ReactCompositeComponent.js?eca4:75)
    at eval (ReactCompositeComponent.js?eca4:728)
```

Thus leading the user to click twice on the open modal button.

**Comment by TheSharpieOne** [comment]
fixed in 2cdf225

**Comment by ericmaicon** [comment]
I am wondering if this is on 5.0.0-alpha.4? If so (I saw the code), it didn't fix for me.

https://github.com/reactstrap/reactstrap/blob/5.0.0-alpha.4/src/Modal.js#L222

Element on that 222 is null after the 2nd time that I try to show the modal. It throws the error because of that.
